### PR TITLE
Update release notes for Core

### PR DESCRIPTION
**Pull request by praveek**
## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:

- [ ] I have signed the [Adobe Open Source CLA](https://opensource.adobe.com/cla.html).
- [ ] My code follows the code style of this project.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [ ] I have read the **CONTRIBUTING** document.
- [ ] I have added tests to cover my changes.
- [ ] All new and existing tests passed.

### CODE REVIEW

**Review by calebk1 — APPROVED** [review]
LGTM!

Though, since they're both being published today, any reason that one is March 13 and the other is March 14?

File: src/pages/documentation/mobile-core/release-notes.md [hunk 1]
```diff
@@ -1,5 +1,18 @@
...
 
+## March 14, 2023
+
+### Android Mobile Core 2.1.0
+
+* Added a new API to the LaunchRulesEngine class that enables adding new rules without replacing the existing ones. 
+
+## March 13, 2023
+
+### iOS AEPCore 3.8.0
```
Comment (swarna04) [inline src/pages/documentation/mobile-core/release-notes.md:11]: nit: `### iOS Mobile Core 3.8.0`

File: src/pages/documentation/release-notes/index.md [hunk 2]
```diff
@@ -4,6 +4,19 @@ description: Release notes and change logs for the Adobe Experience Platform Mob
...
 
+## March 14, 2023
+
+### Android Mobile Core 2.1.0
+
+* Added a new API to the LaunchRulesEngine class that enables adding new rules without replacing the existing ones. 
+
+## March 13, 2023
+
+### iOS AEPCore 3.8.0
```
Comment (swarna04) [inline src/pages/documentation/release-notes/index.md:15]: nit: `### iOS Mobile Core 3.8.0`

**Review by swarna04 — APPROVED** [review]

File: src/pages/documentation/mobile-core/release-notes.md (hunk 1 above)
Comment (praveek) [reply src/pages/documentation/mobile-core/release-notes.md:11]: We have been using AEPCore for iOS until now.

**Comment by praveek** [comment]
> LGTM!
> 
> Though, since they're both being published today, any reason that one is March 13 and the other is March 14?

Yes. We released iOS on 13th and Android on 14th.

File: src/pages/documentation/mobile-core/release-notes.md (hunk 1 above)
Comment (swarna04) [reply src/pages/documentation/mobile-core/release-notes.md:11]: I suppose we should fix it then. We are not using AEP prefixes with any of the other extensions.

Comment (calebk1) [reply src/pages/documentation/mobile-core/release-notes.md:11]: Yeah, consistency for names is IMO better. If all the other ones don't use AEP, this should follow and be iOS Mobile Core instead.

**Review by swarna04 — APPROVED** [review]

**Pull request merged**